### PR TITLE
fix: Update copy for confirmation dialog when initiating a large call - SQSERVICES - 1955

### PR DESCRIPTION
**Pull request by agisilaos**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

We have two issues. The first one is the copy of the alert. And the second issue is that we don't show that alert when trying to initiate a large video call. We only show the alert when trying to initiate an audio call.

### Solutions

With this PR we fix the two issues I mentioned above. In addition to that we make the code in `UIAlertController+OngoingCall.swift` more readable as well as replacing `.localized` with `L10n`.

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: wire-ios/Wire-iOS/Resources/Base.lproj/Localizable.strings [hunk 1]
```diff
@@ -290,8 +290,7 @@
 "conversation.input_bar.audio_message.keyboard.record_tip" = "Tap to record\nYou can  %@  it after that";
 "conversation.input_bar.audio_message.keyboard.filter_tip" = "Choose a filter above";
 
-"conversation.call.many_participants_confirmation.title" = "Start a call";
-"conversation.call.many_participants_confirmation.message" = "This will call %d people";
+"conversation.call.many_participants_confirmation.title" = "Are you sure you want to call %d people?";
```
Comment (johnxnguyen) [inline wire-ios/Wire-iOS/Resources/Base.lproj/Localizable.strings:293]: Just double checking that this copy came from the product / copy team and not from developers?

File: wire-ios/Wire-iOS/Sources/UserInterface/GroupDetails/ConversationActions/ConversationCallController.swift [hunk 2]
```diff
@@ -52,10 +52,24 @@ final class ConversationCallController: NSObject {
...
+    func startVideoCall(started: Completion?) {
+        let startVideoCall = { [weak self] in
+            guard let `self` = self else { return }
+            self.conversation.confirmJoiningCallIfNeeded(alertPresenter: self.target) {
+                started?()
+                self.conversation.startVideoCall()
+            }
+        }
+
+        if conversation.localParticipants.count <= 4 {
```
Comment (netbe) [inline wire-ios/Wire-iOS/Sources/UserInterface/GroupDetails/ConversationActions/ConversationCallController.swift]: use a constant, does this number come from some config too?

File: wire-ios/Wire-iOS/Resources/Base.lproj/Localizable.strings (hunk 1 above)
Comment (agisilaos) [reply wire-ios/Wire-iOS/Resources/Base.lproj/Localizable.strings:293]: Yep, that came directly from Astrid. She left a comment on the ticket.

File: wire-ios/Wire-iOS/Sources/UserInterface/GroupDetails/ConversationActions/ConversationCallController.swift (hunk 2 above)
Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/UserInterface/GroupDetails/ConversationActions/ConversationCallController.swift]: That number came from the product team as well. What would be a good name for that constant?

Comment (netbe) [reply wire-ios/Wire-iOS/Sources/UserInterface/GroupDetails/ConversationActions/ConversationCallController.swift]: what about ConfirmGroupCallParticipantsLimit ?

File: wire-ios/Wire-iOS/Sources/UserInterface/Calling/Helper/UIAlertController+OngoingCall.swift [hunk 3]
```diff
@@ -57,17 +66,17 @@ extension UIAlertController {
...
 
         let controller = UIAlertController(
-            title: effectiveStyle == .alert ? titleKey.localized : messageKey.localized,
-            message: effectiveStyle == .alert ? messageKey.localized : nil,
+            title: effectiveStyle == .alert ? titleKey : messageKey,
+            message: effectiveStyle == .alert ? messageKey : nil,
             preferredStyle: effectiveStyle
         )
-        controller.addAction(.init(title: buttonTitleKey.localized, style: .default) { _ in completion(true) })
+        controller.addAction(.init(title: buttonTitleKey, style: .default) { _ in completion(true) })
```
Comment (johnxnguyen) [inline wire-ios/Wire-iOS/Sources/UserInterface/Calling/Helper/UIAlertController+OngoingCall.swift]: `titleKey`, `messageKey`, and `buttonTitleKey` aren't keys anymore are they? If not, then I suggest to update the parameter names to reflect that.

File: wire-ios/Wire-iOS/Sources/UserInterface/GroupDetails/ConversationActions/ConversationCallController.swift [hunk 4]
```diff
@@ -52,10 +52,24 @@ final class ConversationCallController: NSObject {
...
     }
 
-    func startVideoCall(started: (() -> Void)?) {
-        conversation.confirmJoiningCallIfNeeded(alertPresenter: target) { [conversation] in
-            started?()
-            conversation.startVideoCall()
+    func startVideoCall(started: Completion?) {
+        let startVideoCall = { [weak self] in
+            guard let `self` = self else { return }
+            self.conversation.confirmJoiningCallIfNeeded(alertPresenter: self.target) {
```
Comment (johnxnguyen) [inline wire-ios/Wire-iOS/Sources/UserInterface/GroupDetails/ConversationActions/ConversationCallController.swift:59]: What does `confirmJioningCallIfNeeded` do?

File: wire-ios/Wire-iOS/Sources/UserInterface/GroupDetails/ConversationActions/ConversationCallController.swift (hunk 2 above)
Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/UserInterface/GroupDetails/ConversationActions/ConversationCallController.swift]: Sounds good. I'm going to use that name for the property.

Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/UserInterface/GroupDetails/ConversationActions/ConversationCallController.swift]: Done!

File: wire-ios/Wire-iOS/Sources/UserInterface/GroupDetails/ConversationActions/ConversationCallController.swift (hunk 4 above)
Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/UserInterface/GroupDetails/ConversationActions/ConversationCallController.swift:59]: Here's what it does:

```swift
func confirmJoiningCallIfNeeded(alertPresenter: UIViewController, forceAlertModal: Bool = false, completion: @escaping () -> Void) {
        guard ZMUserSession.shared()?.isCallOngoing == true else {
            return completion()
        }

        let controller = UIAlertController.ongoingCallJoinCallConfirmation(forceAlertModal: forceAlertModal) { confirmed in
            guard confirmed else { return }
            self.endAllCallsExceptIncoming(completion: completion)
        }

        alertPresenter.present(controller, animated: true)
    }
```

File: wire-ios/Wire-iOS/Sources/UserInterface/Calling/Helper/UIAlertController+OngoingCall.swift (hunk 3 above)
Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/UserInterface/Calling/Helper/UIAlertController+OngoingCall.swift]: Done!

**Review by David-Henner — COMMENTED** [review]
should there be new snapshots?

**Comment by agisilaos** [comment]
> should there be new snapshots?

@David-Henner I don't think we have snapshots with that alert. 🤔

**Review by netbe — APPROVED** [review]

**Review by johnxnguyen — APPROVED** [review]

**Pull request merged**